docker_xylem/service.py

```diff
@@ -102,12 +115,30 @@ def unmount_volume(self, request, data):
         path = os.path.join(self.mount_path, name)
 
         try:
-            yield self._umount_fs(path)
-            defer.returnValue({"Err": None})
+            if (yield self._umount_fs(path)):
+                defer.returnValue({"Err": None})
+            else:
+                # Try older mount paths
+                paths = self.get_paths(name)
+                for path in paths:
+                    yield self._umount_fs(path)
+
+                defer.returnValue({"Err": None})
 
         except Exception, e:
             defer.returnValue({"Err": repr(e)})
 
+    def get_paths(self, name):
+        """
+        Function to return an array of old mount paths
+        :param name: Name of volume
+        :return: list of possible paths for given volume name
+        """
+        paths = []
+        for path in self.old_paths:
+            paths.append(os.path.join(path, name))
+        return paths
```

Comment:
This could be replaced by a list comprehension:
`return [os.path.join(path, name) for path in self.old_paths]`